File: BackEnd.v2.1/demo/src/main/java/com/example/demo/models/Battery.java
package com.example.demo.models;


import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;
import java.util.UUID;

@Data
@Entity
@Table(name = "battery")
public class Battery {
    @Id
    @GeneratedValue
    private UUID id;
    private String model;
    private String type;
    private Float storage_capacity;
    private Float nominal_voltage;
    private Float maximum_load_voltage;
    private Float maximum_discharge_voltage;
    private String life_cycle;
    private Float operating_temperature;
    private String lifespan;
    private Date date_manufacture;
    private Float price;
    private Integer quantity;
    @Lob
    @Column(name = "image", length = 500000000)
    private String image;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Company company;
}

File: FrontEnd.v2/src/app/models/battery.ts
/* tslint:disable */

/* eslint-disable */
import {Company} from "./Company";

export interface Battery {
  companyDto: Company;
  date_manufacture: string;
  id?: any;
  image: string;
  life_cycle: string;
  lifespan: string;
  maximum_discharge_voltage: number;
  maximum_load_voltage: number;
  model: string;
  nominal_voltage: number;
  operating_temperature: number;
  price: number;
  quantity: number;
  storage_capacity: number;
  type: string;
}
